app/models/event_invoice.py

```diff
@@ -72,3 +66,8 @@ def get_invoice_number(self):
 
     def __repr__(self):
         return '<EventInvoice %r>' % self.invoice_pdf_url
+
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.identifier=get_new_identifier(EventInvoice)
```

Comment:
This shouldn't be needed

Reply:
Can I make direct call? How should I proceed?